legend-engine-xt-relationalStore-pure/src/main/resources/core_relational/relational/testDataGeneration/testDataGeneration.pure

```diff
@@ -755,13 +755,14 @@ import meta::relational::functions::sqlQueryToString::*;
 
 function meta::relational::testDataGeneration::executionPlan::planSeedDataGenerationWithParameterNameValuePairs(func:FunctionDefinition<Any>[1], mapping:Mapping[1], runtime: Runtime[1], exeCtx: ExecutionContext[1], parameterNameValuePairs: Pair<String, List<Any>>[*], extensions:Extension[*]):ExecutionPlan[1]
 {
-   let relationTree = if($func->functionReturnType().rawType->toOne()->_subTypeOf(TabularDataSet), | scanRelations($func, $mapping, $runtime, $parameterNameValuePairs, $extensions),| scanRelations($func, $mapping, $extensions));
+   let params = $func->functionType().parameters->evaluateAndDeactivate();
+   let newParameterNameValuePairs = $params->size()->range()->map(i | if($params->at($i).genericType.rawType->toOne()->_subTypeOf(Date) && $parameterNameValuePairs->at($i).second.values->size() == 1 && $parameterNameValuePairs->at($i).second.values->at(0)->instanceOf(String),| pair($parameterNameValuePairs->at($i).first, list($parameterNameValuePairs->at($i).second.values->at(0)->cast(@String)->parseDate())),| $parameterNameValuePairs->at($i)));
```

Comment:
Could you format this line into multiple lines so that its easier to read?

Reply:
yea sure! I will clean the code.